charmhelpers/contrib/openstack/amulet/utils.py

```diff
@@ -96,8 +96,8 @@ def validate_endpoint_data(self, endpoints, admin_port, internal_port,
                 'url': self.valid_url,
                 'interface': self.not_null,
                 'service_id': expected['service_id']}
-        return validation_function(endpoints, admin_port, internal_port,
-                                   public_port, expected)
+            return validation_function(endpoints, admin_port, internal_port,
+                                       public_port, expected)
```

Comment:
This looks like a bug?  Are you sure about the additional indent.  If the `if` fails the function will return `None` rather the the v2 endpoint data?

Reply:
Thanks! It is a bug indeed.